### ISSUE TITLE
[Bug] Pulsar Admin sink update can potentially wipe out "inputSpecs"

### ISSUE THREAD

**Issue by aymkhalil** [apache/pulsar#18507]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

2.10 on M1 mac

### Minimal reproduce step

* Create any sink with input specs:
```
bin/pulsar-admin sinks create --name print-sink --archive pulsar-io/data-generator/target/pulsar-io-data-generator-2.10.2.3-SNAPSHOT.nar --inputs persistent://public/default/fresh --input-specs '{"persistent://public/default/fresh":{"receiverQueueSize":10}}'

Created successfully
```
* Try to update the sink with `inputSpecs`
```
bin/pulsar-admin sinks update --name print-sink --inputs persistent://public/default/fresh --input-specs '{"persistent://public/default/fresh":{"receiverQueueSize":1}}'

Updated successfully
```

* Get the effective sink config:
```
bin/pulsar-admin sinks get --name print-sink
```

### What did you expect to see?

```
{
  "tenant": "public",
  "namespace": "default",
  "name": "print-sink",
  "className": "org.apache.pulsar.io.datagenerator.DataGeneratorPrintSink",
  "sourceSubscriptionPosition": "Latest",
  "inputs": [
    "persistent://public/default/fresh"
  ],
  "inputSpecs": {
    "persistent://public/default/fresh": {
      "isRegexPattern": false,
      "schemaProperties": {},
      "consumerProperties": {},
      "receiverQueueSize": 1,           // or update should fail and keep the old value of 10
      "poolMessages": false
    }
  },
  ...
}
```

### What did you see instead?

`receiverQueueSize` is missing from `inputSpecs`
```
{
  "tenant": "public",
  "namespace": "default",
  "name": "print-sink",
  "className": "org.apache.pulsar.io.datagenerator.DataGeneratorPrintSink",
  "sourceSubscriptionPosition": "Latest",
  "inputs": [
    "persistent://public/default/fresh"
  ],
  "inputSpecs": {            
    "persistent://public/default/fresh": {                // receiverQueueSize has been wiped out
      "isRegexPattern": false,
      "schemaProperties": {},
      "consumerProperties": {},
      "poolMessages": false
    } 
  },
  ...
}
```

### Anything else?

If `inputSepcs` updates are not allowed, then the update sink command should not have any side effects. 

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by gaozhangmin** [comment]
I will check this out

**Comment by gaozhangmin** [comment]
@aymkhalil  I can update successfuly with `bin/pulsar-admin sinks update --name print-sink  --input-specs '{"persistent://public/default/fresh":{"receiverQueueSize":12}}'` using master branch

**Comment by aymkhalil** [comment]
@gaozhangmin yes it will update successfully using master, but then if your update have both `--inputs` and `--inputSpecs`, the `consumerConfig` will be reset as per the scenario described in this issues. I attempted a fix here if you want to take a look #19082
